spark/dl/src/main/scala/com/intel/analytics/bigdl/nn/Container.scala

```diff
@@ -41,18 +42,92 @@ abstract class Container[A <: Activity : ClassTag,
     B <: Activity : ClassTag, T: ClassTag](
   implicit ev: TensorNumeric[T]) extends AbstractModule[A, B, T] {
 
+  private var parent: Container[A, B, T] = null
+
   // list of sub modules
   val modules: ArrayBuffer[AbstractModule[Activity, Activity, T]]
   = ArrayBuffer[AbstractModule[Activity, Activity, T]]()
 
+  /**
+   * We base on the order of this returning result for compiling.
+   * In general, it's a list of nodes from topology sort.
+   * return empty as do nothing by default.
+   */
+  private[bigdl] def compilingPath(): List[Node[AbstractModule[Activity, Activity, T]]] = {
```

Comment:
we don't need this method, as container can be convert into a graph by using the toGraph method

Reply:
you mean always convert to `Graph`  first and then `compile` ?

Reply:
I mean in the compilingPath is duplicate with toGraph